### PR TITLE
Change the algorithm for downloading Chromedriver to match the docs

### PR DESCRIPTION
**Pull request by jgraham**
The current code can fail if you have a dev build that's a version
behind the latest version. The new code matches the documentation and
will fall back to an older (likely, but not certainly, compatible)
version if no release-specific version can be found.

### CODE REVIEW

File: tools/wpt/browser.py [hunk 1]
```diff
@@ -548,22 +548,44 @@ def _official_chromedriver_url(self, chrome_version):
...
-            url = "https://storage.googleapis.com/chromium-browser-snapshots/%s/%s/chromedriver_%s.zip" % (
-                self.chromium_platform_string(), revision, self.platform_string())
-            # Check the status without downloading the content (this is a streaming request).
-            get(url)
-        except requests.RequestException:
-            # Fall back to the tip-of-tree Chromium build.
-            revision_url = "https://storage.googleapis.com/chromium-browser-snapshots/%s/LAST_CHANGE" % (
-                self.chromium_platform_string())
+        # The algorithm for downloading Chromedriver is documented at
+        # https://chromedriver.chromium.org/downloads/version-selection
```
Comment (Hexcles) [inline tools/wpt/browser.py]: This is in fact `_official_chromedriver_url`. 

We should instead rework `_latest_chromedriver_url` to make it try `_official_chromedriver_url(current_release)`, `_official_chromedriver_url(current_release-1)`, and then finally `_chromium_chromedriver_url(current_release)`.

**Comment by Hexcles** [comment]
Wait a second, I'm a bit confused. Could you describe what the failure mode is?

If we are using a dev version that's behind the current dev, we should still be able to find the official release of ChromeDriver for it using `_official_chromedriver_url`. The function you're modifying is the fallback when we are using dev version that's too new (without any official release).

**Comment by jgraham** [comment]
Oh I see; somehow I missed `_offical_chromedriver_url`. But that function doesn't go back to earlier versions when there isn't a version for the current major version number. In particular I have Chrome 79 and [https://chromedriver.storage.googleapis.com/LATEST_RELEASE_79]() is 404 at time of writing but [https://chromedriver.storage.googleapis.com/LATEST_RELEASE_78]() isn't.

**Comment by jgraham** [comment]
OK, updated this to just change the `_official_chromedriver_url` function to try more possibilities (and add more logging).

**Review by Hexcles — CHANGES_REQUESTED** [review]
@jgraham is trying the latest Chrome Dev 79, which does not yet have a ChromeDriver release, in which case the code falls back to ToT by design *, which is currently 80.

The [document](https://github.com/web-platform-tests/wpt/blob/master/docs/running-tests/chrome.md) says to use `--channel=dev` in this case, which would suppress the version mismatch error. I agree this is not ideal and am discussing how to make this more ergonomic on IRC.

* the reason I chose to fall back to a newer version of ChromeDriver (e.g. the tip of the tree when everything else fails) is because *empirically* ChromeDriver tends to have better backward than forward compatibility

**Comment by jgraham** [comment]
It seems like this is the wrong approach; filed https://github.com/web-platform-tests/wpt/issues/19899 on the right approach.

**Pull request closed without merging**